app/controllers/application_controller.rb

```diff
@@ -1,5 +1,16 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    @current_user = User.first
+  protect_from_forgery with: :exception
+
```

Comment:
- **[OPTIONAL]** When not signed in users should have no access to the post or user pages. They should be redirected to the sign-in page. To do this you need to authenticate the user by adding the code below to this application controller.

  ```
  before_action :authenticate_user!
  ```